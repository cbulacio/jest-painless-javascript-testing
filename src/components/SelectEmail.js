import React from 'react';

const STATUS = {
	CLICKED: 'clicked',
    NORMAL: 'normal'
};

export default class SelectEmail extends React.Component {
	constructor(props){
		super(props);

        this.state = {
            className: STATUS.NORMAL,
            color:'white'
        }		
	}

	_onClickHandler = () => {
		this.setState({
			className: STATUS.CLICKED,
			color:'lightgreen'
		})
	};

	_onMouseOverHandler = () => {
		this.setState({
			className: STATUS.CLICKED,
			color:'green'
		})
	};

	_onMouseOutHandler = () => {
		this.setState({
			className: STATUS.NORMAL,
			color:'white'
		})
	};

    render (){
        return (<div>
        			<h2>{this.props.title}</h2>
        			<select 
            		className={this.state.className}
            		style={{backgroundColor:this.state.color}}
            		onClick={this._onClickHandler}
            		onMouseOver={this._onMouseOverHandler}
            		onMouseOut={this._onMouseOutHandler}            		
        			autoFocus="true">
                		{this.props.options.map(item =>{
                    		return <option style={{backgroundColor:this.state.color}} 
                    			key={item.id} 
                    			value={item.id}>{item.email}</option>;
                		})}        			
        			</select>
        </div>)
    };

}