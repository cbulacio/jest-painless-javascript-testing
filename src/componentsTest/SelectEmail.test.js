import React from 'react';
import renderer from 'react-test-renderer';

import {users} from '../data/data';
import SelectEmail from '../components/SelectEmail';

it('renders correctly', () =>{
    const tree = renderer
        .create(<SelectEmail title="User emails" options={users} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});


it('changes the background color when click',() => {
    const component = renderer.create(<SelectEmail title="User emails" options={users} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.children[1].props.onClick();

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

it('changes the background color when the mouse is over',() => {
    const component = renderer.create(<SelectEmail title="User emails" options={users} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.children[1].props.onMouseOver();

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

it('changes the background color when the mouse is out',() => {
    const component = renderer.create(<SelectEmail title="User emails" options={users} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.children[1].props.onMouseOut();

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

it('changes in the background color',() => {
    const component = renderer.create(<SelectEmail title="User emails" options={users} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.children[1].props.onClick();

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.children[1].props.onMouseOver();

    tree = component.toJSON();

    expect(tree).toMatchSnapshot();
    tree.children[1].props.onMouseOut();

    tree = component.toJSON();

    expect(tree).toMatchSnapshot();
    
})
