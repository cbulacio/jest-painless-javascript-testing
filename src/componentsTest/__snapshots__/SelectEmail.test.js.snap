// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`changes in the background color 1`] = `
<div>
  <h2>
    User emails
  </h2>
  <select
    autoFocus="true"
    className="normal"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={1}
    >
      Sincere@april.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={2}
    >
      Shanna@melissa.tv
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={3}
    >
      Nathan@yesenia.net
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={4}
    >
      Julianne.OConner@kory.org
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={5}
    >
      Lucio_Hettinger@annie.ca
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={6}
    >
      Karley_Dach@jasper.info
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={7}
    >
      Telly.Hoeger@billy.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={8}
    >
      Sherwood@rosamond.me
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={9}
    >
      Chaim_McDermott@dana.io
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={10}
    >
      Rey.Padberg@karina.biz
    </option>
  </select>
</div>
`;

exports[`changes in the background color 2`] = `
<div>
  <h2>
    User emails
  </h2>
  <select
    autoFocus="true"
    className="clicked"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  >
    <option
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
      value={1}
    >
      Sincere@april.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
      value={2}
    >
      Shanna@melissa.tv
    </option>
    <option
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
      value={3}
    >
      Nathan@yesenia.net
    </option>
    <option
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
      value={4}
    >
      Julianne.OConner@kory.org
    </option>
    <option
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
      value={5}
    >
      Lucio_Hettinger@annie.ca
    </option>
    <option
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
      value={6}
    >
      Karley_Dach@jasper.info
    </option>
    <option
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
      value={7}
    >
      Telly.Hoeger@billy.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
      value={8}
    >
      Sherwood@rosamond.me
    </option>
    <option
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
      value={9}
    >
      Chaim_McDermott@dana.io
    </option>
    <option
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
      value={10}
    >
      Rey.Padberg@karina.biz
    </option>
  </select>
</div>
`;

exports[`changes in the background color 3`] = `
<div>
  <h2>
    User emails
  </h2>
  <select
    autoFocus="true"
    className="clicked"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  >
    <option
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      value={1}
    >
      Sincere@april.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      value={2}
    >
      Shanna@melissa.tv
    </option>
    <option
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      value={3}
    >
      Nathan@yesenia.net
    </option>
    <option
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      value={4}
    >
      Julianne.OConner@kory.org
    </option>
    <option
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      value={5}
    >
      Lucio_Hettinger@annie.ca
    </option>
    <option
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      value={6}
    >
      Karley_Dach@jasper.info
    </option>
    <option
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      value={7}
    >
      Telly.Hoeger@billy.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      value={8}
    >
      Sherwood@rosamond.me
    </option>
    <option
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      value={9}
    >
      Chaim_McDermott@dana.io
    </option>
    <option
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      value={10}
    >
      Rey.Padberg@karina.biz
    </option>
  </select>
</div>
`;

exports[`changes in the background color 4`] = `
<div>
  <h2>
    User emails
  </h2>
  <select
    autoFocus="true"
    className="normal"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={1}
    >
      Sincere@april.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={2}
    >
      Shanna@melissa.tv
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={3}
    >
      Nathan@yesenia.net
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={4}
    >
      Julianne.OConner@kory.org
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={5}
    >
      Lucio_Hettinger@annie.ca
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={6}
    >
      Karley_Dach@jasper.info
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={7}
    >
      Telly.Hoeger@billy.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={8}
    >
      Sherwood@rosamond.me
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={9}
    >
      Chaim_McDermott@dana.io
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={10}
    >
      Rey.Padberg@karina.biz
    </option>
  </select>
</div>
`;

exports[`changes the background color when click 1`] = `
<div>
  <h2>
    User emails
  </h2>
  <select
    autoFocus="true"
    className="normal"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={1}
    >
      Sincere@april.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={2}
    >
      Shanna@melissa.tv
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={3}
    >
      Nathan@yesenia.net
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={4}
    >
      Julianne.OConner@kory.org
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={5}
    >
      Lucio_Hettinger@annie.ca
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={6}
    >
      Karley_Dach@jasper.info
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={7}
    >
      Telly.Hoeger@billy.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={8}
    >
      Sherwood@rosamond.me
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={9}
    >
      Chaim_McDermott@dana.io
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={10}
    >
      Rey.Padberg@karina.biz
    </option>
  </select>
</div>
`;

exports[`changes the background color when click 2`] = `
<div>
  <h2>
    User emails
  </h2>
  <select
    autoFocus="true"
    className="clicked"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "backgroundColor": "lightgreen",
      }
    }
  >
    <option
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
      value={1}
    >
      Sincere@april.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
      value={2}
    >
      Shanna@melissa.tv
    </option>
    <option
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
      value={3}
    >
      Nathan@yesenia.net
    </option>
    <option
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
      value={4}
    >
      Julianne.OConner@kory.org
    </option>
    <option
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
      value={5}
    >
      Lucio_Hettinger@annie.ca
    </option>
    <option
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
      value={6}
    >
      Karley_Dach@jasper.info
    </option>
    <option
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
      value={7}
    >
      Telly.Hoeger@billy.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
      value={8}
    >
      Sherwood@rosamond.me
    </option>
    <option
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
      value={9}
    >
      Chaim_McDermott@dana.io
    </option>
    <option
      style={
        Object {
          "backgroundColor": "lightgreen",
        }
      }
      value={10}
    >
      Rey.Padberg@karina.biz
    </option>
  </select>
</div>
`;

exports[`changes the background color when the mouse is out 1`] = `
<div>
  <h2>
    User emails
  </h2>
  <select
    autoFocus="true"
    className="normal"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={1}
    >
      Sincere@april.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={2}
    >
      Shanna@melissa.tv
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={3}
    >
      Nathan@yesenia.net
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={4}
    >
      Julianne.OConner@kory.org
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={5}
    >
      Lucio_Hettinger@annie.ca
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={6}
    >
      Karley_Dach@jasper.info
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={7}
    >
      Telly.Hoeger@billy.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={8}
    >
      Sherwood@rosamond.me
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={9}
    >
      Chaim_McDermott@dana.io
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={10}
    >
      Rey.Padberg@karina.biz
    </option>
  </select>
</div>
`;

exports[`changes the background color when the mouse is out 2`] = `
<div>
  <h2>
    User emails
  </h2>
  <select
    autoFocus="true"
    className="normal"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={1}
    >
      Sincere@april.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={2}
    >
      Shanna@melissa.tv
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={3}
    >
      Nathan@yesenia.net
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={4}
    >
      Julianne.OConner@kory.org
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={5}
    >
      Lucio_Hettinger@annie.ca
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={6}
    >
      Karley_Dach@jasper.info
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={7}
    >
      Telly.Hoeger@billy.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={8}
    >
      Sherwood@rosamond.me
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={9}
    >
      Chaim_McDermott@dana.io
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={10}
    >
      Rey.Padberg@karina.biz
    </option>
  </select>
</div>
`;

exports[`changes the background color when the mouse is over 1`] = `
<div>
  <h2>
    User emails
  </h2>
  <select
    autoFocus="true"
    className="normal"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={1}
    >
      Sincere@april.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={2}
    >
      Shanna@melissa.tv
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={3}
    >
      Nathan@yesenia.net
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={4}
    >
      Julianne.OConner@kory.org
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={5}
    >
      Lucio_Hettinger@annie.ca
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={6}
    >
      Karley_Dach@jasper.info
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={7}
    >
      Telly.Hoeger@billy.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={8}
    >
      Sherwood@rosamond.me
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={9}
    >
      Chaim_McDermott@dana.io
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={10}
    >
      Rey.Padberg@karina.biz
    </option>
  </select>
</div>
`;

exports[`changes the background color when the mouse is over 2`] = `
<div>
  <h2>
    User emails
  </h2>
  <select
    autoFocus="true"
    className="clicked"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "backgroundColor": "green",
      }
    }
  >
    <option
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      value={1}
    >
      Sincere@april.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      value={2}
    >
      Shanna@melissa.tv
    </option>
    <option
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      value={3}
    >
      Nathan@yesenia.net
    </option>
    <option
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      value={4}
    >
      Julianne.OConner@kory.org
    </option>
    <option
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      value={5}
    >
      Lucio_Hettinger@annie.ca
    </option>
    <option
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      value={6}
    >
      Karley_Dach@jasper.info
    </option>
    <option
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      value={7}
    >
      Telly.Hoeger@billy.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      value={8}
    >
      Sherwood@rosamond.me
    </option>
    <option
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      value={9}
    >
      Chaim_McDermott@dana.io
    </option>
    <option
      style={
        Object {
          "backgroundColor": "green",
        }
      }
      value={10}
    >
      Rey.Padberg@karina.biz
    </option>
  </select>
</div>
`;

exports[`renders correctly 1`] = `
<div>
  <h2>
    User emails
  </h2>
  <select
    autoFocus="true"
    className="normal"
    onClick={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={1}
    >
      Sincere@april.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={2}
    >
      Shanna@melissa.tv
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={3}
    >
      Nathan@yesenia.net
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={4}
    >
      Julianne.OConner@kory.org
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={5}
    >
      Lucio_Hettinger@annie.ca
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={6}
    >
      Karley_Dach@jasper.info
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={7}
    >
      Telly.Hoeger@billy.biz
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={8}
    >
      Sherwood@rosamond.me
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={9}
    >
      Chaim_McDermott@dana.io
    </option>
    <option
      style={
        Object {
          "backgroundColor": "white",
        }
      }
      value={10}
    >
      Rey.Padberg@karina.biz
    </option>
  </select>
</div>
`;
